doc/PROGMEM.rst

```diff
@@ -0,0 +1,266 @@
+Guide to PROGMEM on ESP8266 and Arduino IDE
+===========================================
+
+Intro
```

Comment:
I would start out with putting PROGMEM support in context.  PROGMEM is a Arduino AVR feature that already has documentation on most of the functions, and the Esp8266 support for it follows those guidelines closely.